### DISCUSSION TITLE
Feature Addition: Apparent Intersection point between two polylines2d

### DISCUSSION THREAD

**Discussion by imrankhan1212** [mozman/ezdxf#770] [Ideas]
The current features **ezdxf.math.intersection_line_line_2d** and **ezdxf.math.intersect_polylines_2d**
are good but they give intersection points only if the lines or polylines are intersecting each other. 

In case, when the lines / polylines are not intersecting or appear to be intersecting, the function does not give you any result. So it is requested to add this feature. I am currently using intersection from "Sympy".

This feature is available in pyautocad and is excellent.

Thanks

**Comment by mozman** [comment]
If you mean virtual intersections that don't lie on either line, this is supported and is the default for [`intersection_line_line_2d()`](https://ezdxf.mozman.at/docs/math/core.html#ezdxf.math.intersection_line_line_2d).

```Python
from ezdxf.math import Vec2, intersection_line_line_2d

line0 = Vec2(0, 0), Vec2(1, 0)
line1 = Vec2(2, 1), Vec2(2, 2)

print(intersection_line_line_2d(line0, line1))
```

prints:

    (2.0, 0.0)

Virtual intersections between arbitrary 2D polylines don't seem to be very useful.

**Reply by imrankhan1212** [reply]
intersection between 2 arbitrary polyline2d is actually what I was more interested in.
Anyhow, the library is great. Thanks, and wish you success in future.
